src/test/java/seedu/address/model/concert/UniqueConcertContactListTest.java

```diff
@@ -0,0 +1,191 @@
+        uniqueConcertContactList.add(ALICE_COACHELLA);
+        assertThrows(DuplicateConcertContactException.class, () -> uniqueConcertContactList.add(
+                ALICE_COACHELLA));
+    }
+
+    @Test
+    public void setConcertContact_nullTargetConcert_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.setConcertContact(
+                null, ALICE_COACHELLA));
+    }
+
+    @Test
+    public void setConcertContact_nullEditedConcert_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.setConcertContact(
+                ALICE_COACHELLA, null));
+    }
+
+    @Test
+    public void setConcertContact_targetConcertContactNotInList_throwsConcertNotFoundException() {
+        assertThrows(ConcertContactNotFoundException.class, () -> uniqueConcertContactList
+                .setConcertContact(BOB_COACHELLA, ALICE_COACHELLA));
+    }
+
+    @Test
+    public void setConcertContact_editedConcertIsSameConcert_success() {
+        uniqueConcertContactList.add(ALICE_COACHELLA);
+        uniqueConcertContactList.setConcertContact(ALICE_COACHELLA, ALICE_COACHELLA);
+        UniqueConcertContactList expectedConcertContactList = new UniqueConcertContactList();
+        expectedConcertContactList.add(ALICE_COACHELLA);
+        assertEquals(expectedConcertContactList, uniqueConcertContactList);
+    }
+
+    @Test
+    public void setConcertContact_editedConcertDifferent_success() {
+        uniqueConcertContactList.add(ALICE_COACHELLA);
+        uniqueConcertContactList.setConcertContact(ALICE_COACHELLA, BOB_COACHELLA);
+        UniqueConcertContactList expectedConcertContactList = new UniqueConcertContactList();
+        expectedConcertContactList.add(BOB_COACHELLA);
+        assertEquals(expectedConcertContactList, uniqueConcertContactList);
+    }
+
+    @Test
+    public void remove_nullConcertContact_throwsNullPointerException() {
+        // remove(person)
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.remove(
+                (Person) null));
+
+        // remove(concert)
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.remove(
+                (Concert) null));
+
+        // remove(Person, Concert)
+        assertThrows(NullPointerException.class, () -> uniqueConcertContactList.remove(
+                (Person) null, (Concert) null));
+    }
+
+    @Test
+    public void remove_concertContactNotInList_throwsConcertContactNotFoundException() {
+        assertThrows(ConcertContactNotFoundException.class, () -> uniqueConcertContactList.remove(
+                ALICE, COACHELLA));
+
+        // overloaded methods do not throw
+        assertDoesNotThrow(() -> uniqueConcertContactList.remove(ALICE));
+        assertDoesNotThrow(() -> uniqueConcertContactList.remove(COACHELLA));
+    }
+
```

Comment:
Should add a test for multiple removes. For example, UniqueCCList contains Alice -> Coachella, Alice -> Glastonbury. remove(Alice) should remove both entries.